primitives/src/communities.rs

```diff
@@ -30,7 +32,46 @@ pub type CommunityIndexType = u32;
 pub type LocationIndexType = u32;
 pub type Degree = I64F64;
 pub type NominalIncome = I64F64;
-pub type CommunityIdentifier = H256;
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, Default, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct CommunityIdentifier {
+    location: [u8; 5],
+    bootstrappers: u32,
+}
+
+impl CommunityIdentifier {
+    pub fn new<AccountId: Encode>(location: Location, bootstrappers: Vec<AccountId>) -> CommunityIdentifier {
+        let geo_hash = GeoHash::try_from_params(location.lat, location.lon, 5).unwrap();
+        let crc_engine = Crc::<u32>::new(&CRC_32_CKSUM);
+        let mut digest = crc_engine.digest();
+        digest.update(&(bootstrappers).encode());
+        CommunityIdentifier {
+            location: [
```

Comment:
not idiomatic. do:

let mut geohash_cropped = [0u8; 5];
geohash.clone_from_slice(&geohash[0..5]);

Reply:
or try
let geohash: [u8;5] = Decode::decode(geohash[0..5]);